File: CSharp/estoque/estoque/Entities/Produto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Estoque.Entities
{
    internal class Produto
    {
        private string nome { get; set; }
        private double preco { get; set; }
        private int quantidade;

        public Produto()
        {

        }

        public Produto(string nome, double preco)
        {
            this.nome = nome;
            this.preco = preco;
        }

        public double mostrarTotalEmEstoque()
        {
            return preco * quantidade;
        }

        public void adicionarAoEstoque(int quantidade)
        {
            this.quantidade += quantidade;
        }

        public void removerDoEstoque(int quantidade)
        {
            this.quantidade -= quantidade;
        }

        public override string ToString()
        {
            return nome
                + ", $"
                + string.Format("{0}", preco)
                + ", "
                + quantidade
                + " unidades, Total: $"
                + string.Format("{0}", mostrarTotalEmEstoque());
        }
    }
}

File: PHP/estoque/src/entities/Produto.php
<?php

namespace Deivz\Estoque\entities;

class Produto
{
    private string $nome;
    private float $preco;
    private int $quantidade;

    public function __construct(string $nome, float $preco)
    {
        $this->nome = $nome;
        $this->preco = $preco;
        $this->quantidade = 0;
    }

    public function __get($atributo)
    {
        return $this->$atributo;
    }

    public function mostrarTotalEmEstoque(): float
    {
        return $this->preco * $this->quantidade;
    }

    public function adicionarAoEstoque(int $quantidade): void
    {
        $this->quantidade += $quantidade;
    }

    public function removerDoEstoque(int $quantidade): void
    {
        $this->quantidade -= $quantidade;
    }

    public function __toString()
    {
        return $this->nome
                . ", $"
                . floatval($this->preco)
                . ", "
                . $this->quantidade
                . " unidades, Total: $"
                . $this->mostrarTotalEmEstoque(); 
    }
}